### PR TITLE
Anchor test

### PR DESCRIPTION
**Pull request by EdwinGuzman**
This fixes the old book details page issue for past and future picks.

A link `/books-music-movies/recommendations/best-books/staff-picks/2018-03-01/9780765395726-a-red-peace` should redirect to `/books-music-movies/recommendations/best-books/staff-picks/2018-03-01#9780765395726-a-red-peace`.

An age link `/books-music-movies/recommendations/best-books/staff-picks/2018-01-01/9780316245111-the-end-of-the-wild` should redirect to `/books-music-movies/recommendations/best-books/staff-picks/2018-01-01/9780316245111-the-end-of-the-wild` but also update the picks, filters, and age accordingly.

Removed the `BookPage` component.

### CODE REVIEW

File: test/unit/Main.test.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { shallow, mount } from 'enzyme';
 
 import Main from '../../src/app/components/Application/Main';
 import config from '../../appConfig';
+import Actions from '../../src/app/actions/BookActions';
```
Comment (ktp242) [inline test/unit/Main.test.js]: Eslint complains `'Actions' is defined but never used `

File: test/unit/Main.test.js [hunk 2]
```diff
@@ -58,6 +62,36 @@ describe('Main', () => {
...
+    let component;
+
+    it('should do nothing if the location is wrong', () => {
+      component = mount(
+        <Main
+          listOptions={config.staffPicksListOptions}
+          currentAudience="Adult"
+          location={{ hash: '#01-third-book-title' }}
+          picksData={{ picks }}
+          />
```
Comment (ktp242) [inline test/unit/Main.test.js]: Eslint complains this and line 88 about the closing bracket is not aligned with opening tag.

File: src/app/components/Application/Main.jsx [hunk 3]
```diff
@@ -29,6 +31,29 @@ class Main extends React.Component {
     this.getCount = this.getCount.bind(this);
   }
 
+  componentDidMount() {
+    const picksData = this.props.picksData;
+    const hash = this.props.location && this.props.location.hash ?
+      (this.props.location.hash).substr(1): '';
```
Comment (ktp242) [inline src/app/components/Application/Main.jsx]: 'location' is missing in props validation here

Comment (ktp242) [reply src/app/components/Application/Main.jsx]: It also lacks a space before `:`

**Comment by ktp242** [comment]
Minor issues. Looks good!

**Comment by EdwinGuzman** [comment]
GAH. Eslint broke in my atom environment and now it's harder to see the warnings 😧 .

File: src/app/components/Application/Main.jsx [hunk 4]
```diff
@@ -238,6 +238,7 @@ Main.propTypes = {
   listOptions: PropTypes.object,
   currentSeason: PropTypes.string,
   currentAudience: PropTypes.string,
+  location: PropTypes.shape(PropTypes.object),
```
Comment (ktp242) [inline src/app/components/Application/Main.jsx]: If we use `PropTypes.shape` here,  we should indicate the prop types of its children as the argument. Such as

```
 {
   hash: PropTypes.string,
}
```

**Pull request merged**